cmd/check/main.go

```diff
@@ -35,6 +42,68 @@ func main() {
 		return
 	}
 
+	if req.Source.Ecr {
+		if req.Source.AwsAccessKeyId != "" {
+			os.Setenv("AWS_ACCESS_KEY_ID", req.Source.AwsAccessKeyId)
+		}
+		if req.Source.AwsSecretAccessKey != "" {
+			os.Setenv("AWS_SECRET_ACCESS_KEY", req.Source.AwsSecretAccessKey)
+		}
+		if req.Source.AwsRegion != "" {
+			os.Setenv("AWS_REGION", req.Source.AwsRegion)
+		}
+		mySession := session.Must(session.NewSession())
+		client := ecr.New(mySession)
```

Comment:
I don't know whether it was this line, or lines 62-63 that were causing this to fail earlier on, when the ECR API couldn't be reached.  It'd be nice if we could detect that failure case and spit out a better error message than the JSON-related sympton we we seeing, but I'm not going to block the merge on that.